operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/managed/DependentResourceConfigurator.java

```diff
@@ -1,5 +1,9 @@
 package io.javaoperatorsdk.operator.api.reconciler.dependent.managed;
 
+import java.util.Optional;
+
 public interface DependentResourceConfigurator<C> {
   void configureWith(C config);
+
+  Optional<C> configuration();
```

Comment:
This is meant to retrieve the configuration that was used previously. This implies that the configuration needs to be kept around by implementations so that it can be retrieved at some later time.